### ISSUE TITLE
How to write user agent tests for "Allowed Accessibility Child Roles" / "Required Accessibility Parent Role"

### ISSUE THREAD

**Issue by spectranaut** [w3c/aria#1870]
In #1454, we are introducing a new "User Agents MUST statement":

- "To determine whether an element is the accessibility child of a role, user agents MUST ignore any intervening elements with the role generic or none."
- "To determine whether an element has a parent with the required role, user agents MUST ignore any elements with the role generic or none."

Right now, the only want to test user agent must statements is to test what is exposed in the accessibility API, but I'm struggling to come up with a good test case -- in CORE-AAM there is no logic that depends on whether or not an element has the correct accessibility child or parent.

In html, a table should not be considered a table if it doesn't have the appropriate accessibility children... but HTML-AAM just says to use the WAI-ARIA mappings: https://w3c.github.io/html-aam/#el-table

Maybe this is just a hole in the spec, and html table is a good test case, but am I missing a better ARIA test case?

**Comment by scottaohara** [comment]
resolving these may help on the html aam side... 

https://github.com/w3c/html-aam/issues/293
https://github.com/w3c/html-aam/issues/375

**Comment by spectranaut** [comment]
Hmm I didn't realized there was so much trouble with table already. I'm not sure we can write a test for this, it seems to me these new sections only require validator tests.
